File: Server (ASP.NET Core)/Petadmin/Models/AnimalPageLists.cs
using System.Collections.Generic;
using Petadmin.Core.Models;

namespace Petadmin.Models
{
    public class AnimalPageLists
    {
        public List<string> TypesList { get; set; }
        public List<string> GendersList { get; set; }
        public List<Breed> BreedsList { get; set; }
        public List<string> ColorsList { get; set; }
    }
}

File: Client (Angular)/models/animal-page-lists.ts
import {Type} from './type';
import {Gender} from './gender';
import {Breed} from './breed';
import {Color} from './color';

export interface AnimalPageLists {
  typesList: Type[];
  gendersList: Gender[];
  breedsList: Breed[];
  colorsList: Color[];
}

export class AnimalPageListsEntity implements AnimalPageLists{
  constructor() {
    this.breedsList = [];
    this.colorsList = [];
    this.gendersList = [];
    this.typesList = [];
  }
  breedsList: Breed[];
  colorsList: Color[];
  gendersList: Gender[];
  typesList: Type[];
}
